node/utils.js

```diff
@@ -27,193 +49,251 @@ const reifyVersion = (() => {
+    const absolutePath = path.join(rootPath, relativePath);
+    return isDirectory(absolutePath) || mkdir(absolutePath);
+  }
+  return false;
 }
 
 exports.mkdirp = mkdirp;
 
-function readFileOrNull(filename) {
+function mtime(filepath) {
+  if (useMtimeFastPath) {
+    try {
+      // Used to speed up file stats. Modifies the `statValues` typed array,
+      // with index 11 being the mtime milliseconds stamp. The speedup comes
+      // from not creating Stat objects.
+      if (internalStatValues) {
+        internalStat(filepath);
+      } else {
+        internalStat(filepath, statValues);
+      }
+      return statValues[11];
+    } catch (e) {
+      if (e.code === "ENOENT") {
+        return -1;
+      }
+      useMtimeFastPath = false;
+    }
+  }
+  return fallbackMtime(filepath);
+}
+
+exports.mtime = mtime;
+
+function readdir(dirpath) {
   try {
-    return fs.readFileSync(filename, "utf8");
+    return fs.readdirSync(dirpath);
   } catch (e) {}
   return null;
 }
 
-exports.readFileOrNull = readFileOrNull;
+exports.readdir = readdir;
 
-function readPkgInfo(dir) {
-  const pkg = (() => {
+function readFile(filepath) {
+  if (useReadFileFastPath) {
     try {
-      return dynRequire(path.join(dir, "package.json"));
-    } catch (e) {}
-    return null;
-  })();
+      // Used to speed up reading. Returns the contents of the file as a string
+      // or undefined when the file cannot be opened. The speedup comes from not
+      // creating Error objects on failure.
+      const content = internalModuleReadFile(filepath);
+      return typeof content === "undefined" ? null : content;
+    } catch (e) {
+      useReadFileFastPath = false;
+    }
+  }
+  return fallbackReadFile(filepath);
+}
+
+exports.readFile = readFile;
+
+function readJSON(filepath) {
+  const content = readFile(filepath);
+  return content === null ? content : JSON.parse(content);
+}
 
-  if (! pkg) {
+exports.readJSON = readJSON;
+
+function readPkgInfo(dirpath) {
+  const pkg = readJSON(path.join(dirpath, "package.json"));
+  if (pkg === null) {
     return null;
   }
 
-  const reify = pkg.reify == null ? {} : pkg.reify;
-
+  const reify = pkg.reify;
   if (reify === false) {
     // An explicit "reify": false property in package.json disables
     // reification even if "reify" is listed as a dependency.
     return null;
   }
-
   if (! checkReify(pkg, "dependencies") &&
-      ! checkReify(pkg, "peerDependencies") &&
       // Use case: a package.json file may have "reify" in its
       // "devDependencies" section because it expects another package or
       // application to enable reification in production, but needs its
       // own copy of the "reify" package during development. Disabling
       // reification in production when it was enabled in development
       // would be dangerous in this case.
-      ! checkReify(pkg, "devDependencies")) {
+      ! checkReify(pkg, "devDependencies") &&
+      ! checkReify(pkg, "peerDependencies")) {
```

Comment:
Yep. It was actually because `peerDependencies` are used less so faster fail by checking devDeps first.